### ISSUE TITLE
Copy attention based decoder

### ISSUE THREAD

**Issue by AtmaHou** [IBM/pytorch-seq2seq#153]
You guys really did an excellent work with high scalability. Better than OpenNMT in my opinion :)

Copy mechanism could greatly enhance the performance of Seq2Seq model, would you guys implement a decoder with copy mechanism?

**Comment by pskrunner14** [comment]
@AtmaHou we'll see if it can be integrated in the next release. You're welcome to submit a PR. We need all the help we can get.

**Comment by AtmaHou** [comment]
Thanks for your reply, it will be in great need.

**Comment by varunAI** [comment]
Any Update on this?

**Comment by pskrunner14** [comment]
@varunAI Yup working on it. Will push to develop as soon as it's done.

**Comment by rrkarim** [comment]
@pskrunner14 I can work on this if you need some help.

**Comment by diegoantognini** [comment]
@pskrunner14 Any news on this ?

**Comment by diegoantognini** [comment]
@pskrunner14 is this PR https://github.com/IBM/pytorch-seq2seq/pull/172 reliable for the copy mechanism ?

**Comment by pskrunner14** [comment]
@Diego999 no, I'm facing some issues with dimensions as mentioned and there seems to be a problem with the dynamic dictionary too. I'll get to this asap. Will post any updates on this thread.

**Comment by diegoantognini** [comment]
@pskrunner14 Thank you for your quick answer !

**Comment by diegoantognini** [comment]
@pskrunner14 Do you have any news about it ? Thanks
